rfcs/0030-review-workflow.md

```diff
@@ -0,0 +1,147 @@
+to explain this process to first time contributors.
+
+## Advantages
+
+All the above issues should be resolved by this:
+
+1. There is no reason anymore to preferably review recent PRs. Before, recent
+   commit were the only ones where one can be sure that they are not reviewed
+   yet (and thus in a `needs:work` state). Now, reviewers can randomly choose
+   and `needs:review` PR.
+
+2. Similarly, the reviewer may change at any time. If the first reviewer forgets
+   or loses interest in a PR, the PR will just end up in the `needs:review` list
+   again and can easily find a new reviewer.
+
+3. People without commit access can help reviewing commits and lessen the load
+   on committers without putting the reviewed PRs at a disadvantage.
+
+4. We can easily gather statistics. In the long run, we might even implement
+   some sort of automated triage where we regularly ping `needs:work` PRs and
+   close them after the authors went unresponsive for a while. Or we could
+   tag them as `needs:takeover`. But this is out of scope.
+
+
+An example PR might go like this. `contributor` creates the PR, `reviewer1`
+reviews without commit access, `reviewer2` reviews with commit access.
+
+> contributor: hello: 42.0 -> 43.0 [PR text]
+
+[the bot automatically sets the new PR to `needs:review`]
+
+> reviewer1: Looks good to me. Just one little nitpick: Please add a waving emoji to the name.
+> @GrahamcOfBorg status needs:work
+
+> contributor: Good point, more emojis are always good. Done.
+> @GrahamcOfBorg status needs:review
+
+> reviewer1: Thanks! I think this is ready for merge.
+> @GrahamcOfBorg status needs:merge
+
+> reviewer2: LGTM
+> @GrahamcOfBorg build-and-merge
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Contributing would be a bit more complicated. There is one more bit of
+  knowledge new contributors need. It is possible that some contributors forget
+  to set the PR to `needs:review` again after a fixup. That might be resolved
+  by some form of automatic triage as explained above. It would also be not much
+  than the current status, since the first reviewer will still be notified by
+  GitHub.
+
+- We rely even more on the tooling than we do now. However the general workflow
+  should be portable to different tooling and might even work better there.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Move to a different issue tracker. That would be a great effort and most
+  more powerful issue trackers would also be unfamiliar to new contributors,
+  discouraging drive-by contributors.
+
+- Do nothing. All the issues mentioned in the motivation remain unsolved.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- List the exact states. We should probably start simple and evolve from there.
+
+- Finish the tooling.
+
+- Write documentation. It should at least be explained in `CONTRIBUTING.md`,
+  maybe a greeting bot explaining it to first time contributors would be even
+  better.
```

Comment:
I'd actually do both things.

Reply:
Agree. Once we decide we choose a tooling / start implement this, we should list this as a requirement.